src/v/datalake/translation/partition_translator.cc

```diff
@@ -197,65 +222,77 @@ ss::future<> partition_translator::translate_until_stopped() {
         // Update partition metrics. Note that last committed offset here
         // is NOT synchronized with outstanding commit operations. Therefore
         // if we reach this point before the most recent batch of files has
-        // been committed, the commit lag metric will be out of sync at least
-        // until 'wait_for_data' returns and we re-enter the loop.
+        // been committed, the commit lag metric will be out of sync at
+        // least until 'wait_for_data' returns and we re-enter the loop.
         _data_source->update_commit_lag(last_committed_offset);
         _data_source->update_translation_lag(last_translated_offset);
 
+        auto local_last_translated_offset = last_translated_offset;
+        if (
+          auto last_known_translated_offset
+          = _translation_ctx->last_translated_offset()) {
+            // A translation already in progress, start from where we left off.
+            local_last_translated_offset = last_known_translated_offset.value();
+        }
+
+        static constexpr auto data_wait_duration = 3s;
         // Wait until some data is ready to be translated.
         auto maybe_begin_offset
           = co_await _data_source->wait_for_data_to_translate(
-            last_translated_offset, _as);
+            local_last_translated_offset,
+            ss::lowres_clock::now() + data_wait_duration,
+            _as);
 
-        // if wait_for_data timed out (i.e. all trnaslatable records have
+        // if wait_for_data timed out (i.e. all translatable records have
         // been translated already), reenter the loop. this gives us an
         // opportunity to reconcile outstanding coordinator state, which is
         // helpful for keeping lag metrics up-to-date.
-        if (!maybe_begin_offset.has_value()) {
-            continue;
+        if (maybe_begin_offset) {
+            auto begin_offset = maybe_begin_offset.value();
+
+            // Notify the scheduler that there is some data to translate
+            _scheduler->notify_ready(id);
```

Comment:
is it ok to notify that we are ready for translation before finishing the previous one

Reply:
no, its not. The call becomes a no-op, logs a warning and exits. A translator can only be in one state at a time. (scheduler expects a translator to be well behaved in some sense).